cli/src/cluster_query.rs

```diff
@@ -106,6 +106,18 @@ impl ClusterQuerySubCommands for App<'_, '_> {
                         .takes_value(false)
                         .help("Continue reporting progress even after the validator has caught up"),
                 )
+                .arg(
+                    Arg::with_name("our_localhost")
```

Comment:
We could probably do away with the flag by instead cycling between checking gossip for contact info and successfully querying localhost in the case that `node_json_rpc_url` wasn't supplied.  WDYT?

Reply:
using gossip is nice idea. :+1: 

But I'd like it to be separate option (`--use-gossip`) for following reasons:

- newly depending on gossip protocol might needlessly complicate trouble shooting. Also, we have managed not hard-coding the mainnet-beta entrypoint up to now in our code base. That means using gossip would also require `--entrypoint` to `catchup` to override.
- the most often observed use case of `catchup` is running it locally for locally-running validator for a particular subset of partners, so I want to just support the use-case well without waiting for gossip discovery (which takes some time). In that case, preferably I want `node_identity` to be omitted along with `node_json_rpc_url`. And make `catchup --our-localhost` dead-simple and just-work.
- btw, I also considered changing `solana catchup` (no option) behaviour to `solana catchup --our-localhost`. But, the general `solana`'s behavior is access to url in config or api.mainnet-beta if not set by config. So, it sounded odd for `solana catchup` to suddenly accessing localhost:8899 if no option given.

Reply:
Ah!  Sorry I miscommunicated here.  I meant alternate between gossip contact info like we do today (via RPC) and a successful localhost RPC query.